src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1270,13 +1270,42 @@ private static void LinkBlocks(BasicBlockBuilder prevBlock, BasicBlockBuilder ne
 
         public override IOperation VisitBlock(IBlockOperation operation, int? captureIdForResult)
         {
-            Debug.Assert(_currentStatement == operation);
+            StartVisitingStatement(operation);
 
             EnterRegion(new RegionBuilder(ControlFlowRegionKind.LocalLifetime, locals: operation.Locals));
             VisitStatements(operation.Operations);
             LeaveRegion();
 
-            return null;
+            return FinishVisitingStatement(operation);
+        }
+
+        private void StartVisitingStatement(IOperation operation)
+        {
+            Debug.Assert(_currentStatement == operation);
+            Debug.Assert(_evalStack.Count == 0);
+            SpillEvalStack();
+        }
+
+        private IOperation FinishVisitingStatement(IOperation originalOperaion, IOperation result = null)
```

Comment:
The fact that we only need the result parameter in 2 cases feels like that means that we should just call `AddStatement` there, and return null in all cases. We're asserting in `StartVisitingStatement` that the `if` below should always be taken, so I don't see any big problems with this, and it will make our implementation more standard.